pkg/server/node/manager_test.go

```diff
@@ -0,0 +1,163 @@
+package node
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/spire/pkg/server/datastore"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/test/clock"
+	"github.com/spiffe/spire/test/fakes/fakedatastore"
+	"github.com/spiffe/spire/test/fakes/fakemetrics"
+	"github.com/spiffe/spire/test/spiretest"
+)
+
+func TestManager(t *testing.T) {
+	spiretest.Run(t, new(ManagerSuite))
+}
+
+type ManagerSuite struct {
+	spiretest.Suite
+
+	clock   *clock.Mock
+	log     logrus.FieldLogger
+	logHook *test.Hook
+	ds      *fakedatastore.DataStore
+	metrics *fakemetrics.FakeMetrics
+
+	m *Manager
+}
+
+func (s *ManagerSuite) SetupTest() {
+	s.clock = clock.NewMock(s.T())
+	s.log, s.logHook = test.NewNullLogger()
+	s.ds = fakedatastore.New(s.T())
+	s.metrics = fakemetrics.New()
+}
+
+func (s *ManagerSuite) TestPruning() {
+	expiredFor := 5 * time.Minute
+
+	done := s.setupAndRunManager(expiredFor)
+	defer done()
+
+	// banned node is never pruned
+	nodeBanned := &common.AttestedNode{
+		SpiffeId:            "spiffe://test.test/banned",
+		AttestationDataType: "aws-tag",
+		CertSerialNumber:    "",
+		CanReattest:         true,
+		CertNotAfter:        s.clock.Now().Unix(),
+	}
+	attestedNodeBanned, err := s.ds.CreateAttestedNode(context.Background(), nodeBanned)
+	s.NoError(err)
+
+	// non-reattestable node is pruned when IncludeNonReattestable == true
+	nodeNonReattestable := &common.AttestedNode{
+		SpiffeId:            "spiffe://test.test/tofu",
+		AttestationDataType: "aws-tag",
+		CertSerialNumber:    "badcafe",
+		CanReattest:         false,
+		CertNotAfter:        s.clock.Now().Unix(),
+	}
+	attestedNodeNonReattestable, err := s.ds.CreateAttestedNode(context.Background(), nodeNonReattestable)
```

Comment:
Can you get the context from T and use it instead of context.Background() in all tests?

```suggestion
        ctx := s.T().Context()

	attestedNodeNonReattestable, err := s.ds.CreateAttestedNode(ctx, nodeNonReattestable)
```